File: frontend/src/types/warehouse/warehouseBin.ts
export interface WarehouseBin {
  id: number
  binCode: string
  rowNo: string
  colNo: number | null
  levelNo: number | null
  remark: string | null
  operatorUserId: number | null
  operatorName: string | null
  createdAt: string
  updatedAt: string
}

export interface WarehouseBinQuery {
  binCode?: string
  rowNo?: string
  colNo?: number
  levelNo?: number
  page?: number
  pageSize?: number
}

export interface WarehouseBinSavePayload {
  rowNo: string
  colNo?: number | null
  levelNo?: number | null
  remark?: string | null
}

export type WarehouseBinExportQuery = Omit<WarehouseBinQuery, 'page' | 'pageSize'> & {
  ids?: number[]
}

File: backend/src/main/java/com/storage/warehouse/entity/WarehouseBin.java
package com.storage.warehouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("warehouse_bin")
public class WarehouseBin {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String binCode;

    private String rowNo;

    private Integer colNo;

    private Integer levelNo;

    private String remark;

    private Long operatorUserId;

    private String operatorName;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
